provider/spark.go

```diff
@@ -556,15 +656,45 @@ func (spark *SparkOfflineStore) CreatePrimaryTable(id ResourceID, schema TableSc
 }
 
 func (spark *SparkOfflineStore) GetPrimaryTable(id ResourceID) (PrimaryTable, error) {
-	return nil, nil
+	path := fmt.Sprintf("%sresource.parquet", ResourcePrefix(id))
+	table, err := spark.Store.DownloadParquetTable(path)
+	if err != nil {
+		return nil, err
+	}
+	tableList := table.([]interface{})
+	var sourcePath string
+	for _, v := range tableList {
+		sourcePath = reflect.ValueOf(v).FieldByName("Source").String()
+	}
+	return &S3PrimaryTable{spark.Store, sourcePath}, nil
 }
 
 func (spark *SparkOfflineStore) CreateResourceTable(id ResourceID, schema TableSchema) (OfflineTable, error) {
 	return nil, nil
 }
 
 func (spark *SparkOfflineStore) GetResourceTable(id ResourceID) (OfflineTable, error) {
-	return nil, nil
+	path := fmt.Sprintf("%sresource.parquet", ResourcePrefix(id))
+	table, err := spark.Store.DownloadParquetTable(path)
+	if err != nil {
+		return nil, err
+	}
+	list := reflect.ValueOf(table).Index(0)
```

Comment:
This is unclear. Can it be in a function with a descriptive name? Otherwise comment what its doing

Reply:
added more descriptive name